src/main/java/com/example/petbuddybackend/service/user/CaretakerService.java

```diff
@@ -97,29 +96,27 @@ public RatingResponse deleteRating(String caretakerEmail, String clientEmail) {
     public CaretakerComplexInfoDTO addCaretaker(CreateCaretakerDTO caretaker, String email) {
         assertCaretakerNotExists(email);
         AppUser appUser = userService.getAppUser(email);
-        Caretaker caretakerToSave = caretakerMapper.mapToCaretaker(caretaker, appUser);
+        Caretaker mappedCaretaker = caretakerMapper.mapToCaretaker(caretaker, appUser);
 
-        PhotoLink profilePicture = photoService.findByNullableId(appUser.getProfilePictureBlob()).orElse(null);
-        caretakerToSave = caretakerRepository.save(caretakerToSave);
-
-        return caretakerMapper.mapToCaretakerComplexInfoDTO(caretakerToSave, profilePicture);
+        renewCaretakerProfilePicture(mappedCaretaker);
```

Comment:
nie wiem czy od razu nie wywolac tego:
```suggestion
        userService.renewProfilePicture(caretaker.getAccountData());
```

Reply:
albo nawet wywolac bezposrednio metode z PhotoService

Reply:
Chyba przy userService zostane. Tam jeszcze jest check czy użytkownik faktycznie ma jakieś zdjęcie profilowe

Reply:
ma sens